src/Traits/HasMockClient.php

```diff
@@ -25,6 +29,49 @@ public function withMockClient(MockClient $mockClient): static
         return $this;
     }
 
+    /**
+     * Creates a MockResponse instance from the given value if
+     * it is a string or an array. Otherwise, it returns the
+     * value as is.
+     */
+    private function prepareMockResponse(mixed $value): MockResponse|Fixture|callable
+    {
+        if (is_array($value) || is_string($value)) {
+            return new MockResponse($value);
+        }
+
+        return $value;
+    }
+
+    /**
+     * Mocks the given requests with the given responses, or with the
+     * mocked response from the 'getFakeData()' method if the
+     * request has implemented the HasFakeData interface.
+     */
+    public function withRequestMocks(array $requestMocks): static
+    {
+        $responses = collect($requestMocks)->mapWithKeys(function (mixed $value, int|string $key) {
+            // Something like: [UserRequest::class => MockResponse::fixture('user')];
+            // Or URL mocking: ['/user' => ['id' => 123];
+            if (! is_numeric($key)) {
```

Comment:
I mainly used `is_numeric` because that's what is used throughout the Laravel codebase: https://github.com/laravel/framework/blob/38ae673608b334c983ba7e729e605eef9729eab8/src/Illuminate/Collections/Arr.php#L829